lib/pure/unicode.nim

```diff
@@ -20,12 +22,29 @@ type
 
 {.deprecated: [TRune: Rune, TRune16: Rune16].}
 
+proc `<=`(a, b: Rune): bool = return a.int32 <= b.int32
+proc `<=`(a: RuneImpl, b: Rune): bool = return a.int32 <= b.int32
+proc `<=`(a: Rune, b: RuneImpl): bool = return a.int32 <= b.int32
+proc `==`(a: RuneImpl, b: Rune): bool = return a.int32 == b.int32
+proc `==`(a: Rune, b: RuneImpl): bool = return a.int32 == b.int32
+
 proc `<=%`*(a, b: Rune): bool = return int(a) <=% int(b)
 proc `<%`*(a, b: Rune): bool = return int(a) <% int(b)
 proc `==`*(a, b: Rune): bool = return int(a) == int(b)
 
 template ones(n: untyped): untyped = ((1 shl n)-1)
 
+proc runeLen*(r: Rune): int {.noSideEffect.} =
```

Comment:
Shouldn't this simply be called ``len``?

Reply:
@dom96 Hmm... as stated in [API naming design](https://nim-lang.org/docs/apis.html) I'd argue that is should actually be called `size`.

Reply:
I still vote for ``len``. ``runeLenAt`` works on a ``string``. This is a proc that takes a custom type ``Rune``. The convention is to use ``len``.